### PR TITLE
feature/backend/save-jobs-2

### PR DESCRIPTION
**Pull request by katharinaviik**
# Title  
**Implement Saved Jobs Feature (Backend)**  

---

### **Saved Jobs Module Overview**
- **Endpoints:**  
  - `GET /api/:user_id/saved` → View saved jobs  
  - `POST /api/:user_id/saved` → Add a job to saved list  
  - `DELETE /api/:user_id/saved` → Remove a job from saved list  
- **Database:**  
  - Added `SavedJob` model linking `User` ↔ `JobPost`
  - Enforced unique constraint per user–job pair  
- **Permissions:**  
  - Only authenticated students can save or unsave jobs  
- **Integration:**  
  - Includes job details serialization (title, company, location, salary)
  - Works with frontend student dashboard “Saved Jobs” tab
- **Testing:**  
  - Unit and integration tests for CRUD operations  
  - Auth and edge cases covered (duplicate saves, invalid job IDs)
- **Documentation:**  
  - Updated API reference and Postman collection  

---

## Functional Acceptance Criteria  

- [x] Student can **view** their saved jobs list  
- [x] Student can **add** a job to their saved list  
- [x] Student can **remove** a job from their saved list  
- [x] Duplicate saves are prevented  
- [x] Only authenticated users can access their saved jobs  
- [x] 100% tests passing for saved jobs module  

---

## Checklist  

- [x] Self-review completed  
- [x] Updated API documentation  
- [x] Added unit and integration tests  
- [x] Code follows Backend Coding Guidelines  
- [x] All migrations applied successfully  

---

## Type of Change  

- [ ] 🐞 Bug fix  
- [x] ✨ New feature  
- [ ] 📝 Documentation update  
- [ ] 🔧 Other (please specify)  

---

## Additional Information  

**Breaking change?** No  
**Migrations:** 1 new model (`SavedJob`)  
**Endpoints added:** `/api/:user_id/saved/` (GET, POST, DELETE)  
**Related branches:** `frontend/save-jobs`, `dev`

### CODE REVIEW

File: backend/src/routes/:user_id/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  deleteSaved,
+} = require("../../controllers/savedController");
+
+// Routes are mounted at /:user_id by routes registrar
+// So these define /:user_id/saved
+
+/**
+ * GET /:user_id/saved
+ */
+router.get("/saved", [validateUserId, handleValidationResult], getSaved);
```
[the remark this answers is not in the document]
Comment (katharinaviik) [reply backend/src/routes/:user_id/index.js]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/AvengerDisassemble/KU-connect/pull/278#discussion_r2488958784)

File: backend/src/routes/:user_id/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * POST /:user_id/saved
+ */
+router.post(
+  "/saved",
+  [validateUserId, validateJobIdInBody, handleValidationResult],
+  postSaved,
+);
```
[the remark this answers is not in the document]
Comment (katharinaviik) [reply backend/src/routes/:user_id/index.js]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/AvengerDisassemble/KU-connect/pull/278#discussion_r2488958792)
